### DISCUSSION TITLE
Remote File Operations Possible?

### DISCUSSION THREAD

**Discussion by lbeckman314** [minio/minio#18877] [Q&A] [answered]
Hello, we are attempting to upload files from a remote server (non-S3) to a MinIO bucket and were wondering if the following use case was possible:

**Remote File Upload Example:** ❓
```sh
mc cp user@remote-server:/remote-data.txt <minio-bucket>
```

We didn't see any [definitive answer](https://min.io/docs/minio/linux/search.html?minio%5Bquery%5D=remote+server&minio%5BrefinementList%5D%5Bplatform%5D%5B0%5D=linux) on this use case in the docs but very well might have overlooked it. Thank you!

---

**Possible Alternatives:**
1. Install `mc` on the remote server and simply SSH and run `mc` file operations there.
2. Install a MinIO (S3) server on the remote filesystem that we want to upload, then run `mc` commands locally after registering the remote MinIO bucket.
3. Install an [FTP server](https://min.io/docs/minio/linux/developers/file-transfer-protocol.html) on the remote filesystem, then run `mc` commands locally.

**Comment by klauspost** [comment, marked as answer]
Option 1.

I have no idea how your suggestion would be supposed to work, but option 1. is it.

**Reply by lbeckman314** [reply]
We'll proceed with option 1 then, thank you! The example command above might satisfy the case where: 
1. Files to be uploaded are located on a semote Server
2. SSH access is enabled for the Server
3. `mc` is not yet installed on the Server (and restrictions on installing software may be in place)

Not sure how common this setup might be, in any case thank you for the clarification!
